### PR TITLE
support default value (IR only)

### PR DESCRIPTION
**Pull request by Eggiverse**
Add support for default value.

For now, default value lives only in IR.

It is considered helpful for a GUI.

### CODE REVIEW

File: src/codegen/codegen.jl [hunk 1]
```diff
@@ -5,7 +5,7 @@ using MatchCore
 using Comonicon.Types
 
 export ASTCtx, ZSHCompletionCtx
-export codegen, rm_lineinfo, prettify, ASTCtx, pushmaybe!
```
Comment (Eggiverse) [inline src/codegen/codegen.jl:8]: Remove repeated export

File: src/parse/cast.jl [hunk 2]
```diff
@@ -123,29 +123,29 @@ function parse_kwargs(def)
 
     # (name, type, isflag)
     for each in def[:kwargs]
-        push!(types.args, Expr(:tuple, to_option_or_flag(each)...))
+        push!(types.args, to_option_or_flag(each))
```
Comment (Eggiverse) [inline src/parse/cast.jl:126]: The most controversial part.

Without this change, default value is not parsed as expression if it is given in that way.

**Review by Eggiverse — COMMENTED** [review]
Some comments

File: src/parse/cast.jl [hunk 3]
```diff
@@ -129,23 +129,23 @@ function parse_kwargs(def)
...
-    # (name, type, require, vararg)
+    # (name, type, require, vararg, default_value)
     @smatch ex begin
-        ::Symbol => (string(ex), Any, true, false)
-        :($name::$type) => (string(name), wrap_type(def, type), true, false)
-        :($name::$type...) => (string(name), wrap_type(def, type), true, true)
-        Expr(:kw, :($name::$type), _) => (string(name), wrap_type(def, type), false, false)
-        :($name...) => (string(name), Any, true, true)
-        Expr(:kw, name::Symbol, _) => (string(name), Any, false, false)
+        ::Symbol => (string(ex), Any, true, false, nothing)
```
Comment (Roger-luo) [inline src/parse/cast.jl:134]: I think we need to use `Symbol` for the name if we are recording the default value. Given potentially we will want to replicate the julia function behaviour by inferring variable dependencies in the default value. So things like `f(x; y=sin(x))` will work.

Comment (Roger-luo) [reply src/parse/cast.jl:134]: but need to check other places when changing this.

**Comment by Roger-luo** [comment]
Is this ready for review?

**Comment by Eggiverse** [comment]
It is good enough for simply showing default value (as an expression when necessary).

More functionalities can be left for future.

**Comment by Eggiverse** [comment]
Default value validation is added.

It is probably not perfect but it should cover most cases.

**Comment by Eggiverse** [comment]
It serves as a hint only

**Comment by Roger-luo** [comment]
nice work! thanks!

**Pull request merged**